engine/src/main/java/org/hibernate/search/engine/metadata/impl/PropertyMetadata.java

```diff
@@ -64,11 +64,18 @@ public String getPropertyAccessorName() {
 		return propertyAccessorName;
 	}
 
-	// TODO HSEARCH-1867 change return type to set
+	/**
+	 * @deprecated use getFieldMetadataSet() instead
+	 */
+	@Deprecated
 	public List<DocumentFieldMetadata> getFieldMetadata() {
 		return new ArrayList<DocumentFieldMetadata>( documentFieldMetadataList );
 	}
 
+	public Set<DocumentFieldMetadata> getFieldMetadataSet() {
```

Comment:
Why did you not just change the return type of the existing method? Please don't tell me you are trying to keep backwards compat for internal types again ;)